akc/provider.go

```diff
@@ -18,12 +44,34 @@ func Provider() *schema.Provider {
 			"akc_key_value":  dataSourceKeyValue(),
 			"akc_key_secret": dataSourceKeySecret(),
 		},
+		ConfigureFunc: providerConfigure,
 	}
 }
 
-// ProviderConfigure is only here for starting with a good template
-func ProviderConfigure(d *schema.ResourceData) error {
-	log.Printf("[DEBUG] configure")
+func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	endpoint := d.Get("endpoint").(string)
+	clientId := d.Get("client_id").(string)
+	clientSecret := d.Get("client_secret").(string)
+	tenantId := d.Get("tenant_id").(string)
+	authMethod := d.Get("auth_method").(string)
+
+	if (clientId != "") && (clientSecret != "") {
+		authMethod = "creds"
+	}
+
+	var c *client.Client
+	var err error
 
-	return nil
+	switch authMethod {
+	case "creds":
+		c, err = client.NewClientCreds(endpoint, clientId, clientSecret, tenantId)
+	case "cli":
+		c, err = client.NewClientCli(endpoint)
+	case "msi":
+		c, err = client.NewClientMsi(endpoint)
+	case "env":
+	default:
+		c, err = client.NewClientEnv(endpoint)

```

Comment:
Maybe we could default to `NewClientCli` ?

Reply:
let me fix it up to follow the same as azurerm:
- `msi` (if `msi=true`)
- `creds` (if set)
- `env` (if set)
- `cli` (default)

Reply:
@arkiaconsulting  done